src/components/ListItem.jsx

```diff
@@ -1,17 +1,21 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState } from 'react';
 import './ListItem.css';
 import { updateItem } from '../api/firebase';
 
 export function ListItem({ item, listPath }) {
 	const { id, totalPurchases, name, dateLastPurchased } = item;
-	const lessThan24HoursSincePurchase =
-		dateLastPurchased &&
-		(new Date() - dateLastPurchased.toDate()) / 3600000 < 24;
+
+	const isLessThan24HoursSinceLastPurchased =
+		compareIfDateIsLessThan24Hours(dateLastPurchased);
 
 	const [isChecked, setIsChecked] = useState(
-		lessThan24HoursSincePurchase || false,
+		isLessThan24HoursSinceLastPurchased || false,
 	);
 
+	function compareIfDateIsLessThan24Hours(date) {
+		return date && (new Date() - date.toDate()) / 3600000 < 24;
```

Comment:
EXCELLENT place for a comment to explain what’s going on here (what are the magic numbers? - you know, and I know, right now, but what about 6 months from now?)